src/modules/dashboard/components/Dialogs/AddExistingSafeDialog/AddExistingSafeDialogForm.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import { FormikProps } from 'formik';
+import { defineMessages } from 'react-intl';
+import Button from '~core/Button';
+import { ActionDialogProps } from '~core/Dialog';
+import DialogSection from '~core/Dialog/DialogSection';
+import { Select, Input, Annotations } from '~core/Fields';
+import Heading from '~core/Heading';
+import { chains } from './gnosisSafeChains';
+
+import { FormValues } from './AddExistingSafeDialog';
+
+import styles from './AddExistingSafeDialogForm.css';
+
+const MSG = defineMessages({
+  title: {
+    id: 'dashboard.AddExistingPaymentDialog.AddExistingSafeDialogForm.title',
+    defaultMessage: 'Add Gnosis Safe',
+  },
+  chain: {
+    id: 'dashboard.AddExistingPaymentDialog.AddExistingSafeDialogForm.chain',
+    defaultMessage: 'Select Chain',
+  },
+  safeName: {
+    id: 'dashboard.AddExistingPaymentDialog.AddExistingSafeDialogForm.safeName',
+    defaultMessage: 'Name the safe',
+  },
+  annotation: {
+    id:
+      'dashboard.AddExistingPaymentDialog.AddExistingSafeDialogForm.annotation',
+    defaultMessage: 'Explain why you’re adding this Gnosis Safe (optional)',
+  },
+  contract: {
+    id: 'dashboard.AddExistingPaymentDialog.AddExistingSafeDialogForm.contract',
+    defaultMessage: 'Add Safe Contract',
+  },
+  safeFound: {
+    id:
+      'dashboard.AddExistingPaymentDialog.AddExistingSafeDialogForm.safeFound',
+    // @TODO remove hardcoding here
+    defaultMessage: 'Safe found on Gnosis Chain',
+  },
+});
+interface Props extends ActionDialogProps {
+  ethDomainId?: number;
+}
+
+const AddExistingSafeDialogForm = ({
+  back,
+  handleSubmit,
+  isSubmitting,
+  isValid,
+}: Props & FormikProps<FormValues>) => {
+  return (
+    <>
+      <DialogSection appearance={{ theme: 'sidePadding' }}>
+        <div className={styles.headingContainer}>
+          <Heading
+            appearance={{ size: 'medium', margin: 'none', theme: 'dark' }}
+            text={MSG.title}
+          />
+        </div>
+      </DialogSection>
+      <DialogSection>
+        <div className={styles.chainSelect}>
+          <Select
+            options={chains}
+            label={MSG.chain}
+            name="chainId"
+            appearance={{ theme: 'grey', width: 'fluid' }}
+            disabled={isSubmitting}
+          />
+        </div>
+      </DialogSection>
+      <DialogSection>
+        <Input
+          name="contractAddress"
+          label={MSG.contract}
+          appearance={{ colorSchema: 'grey', theme: 'fat' }}
+          disabled={isSubmitting}
+          status={MSG.safeFound}
+        />
+      </DialogSection>
+      <DialogSection>
+        <Input
+          label={MSG.safeName}
+          name="safeName"
+          appearance={{ colorSchema: 'grey', theme: 'fat' }}
+          disabled={isSubmitting}
+          maxLength={20}
```

Comment:
I think this case we'll have to adapt to the length Gnosis accepts for their safe names, rather then imposing our own

So either change it to the "correct one" _(you'll have to find it out yourself, or leave a `TODO` comment for later

Reply:
I looked into this, because Gnosis Safes are locally stored names, they don't seem to impose a limit on them. So, we can impose our own limit here.

Reply:
I looked into this, because Gnosis Safes are locally stored names, they don't seem to impose a limit on them. So, we can impose our own limit here.